### PR TITLE
feat(benchtop): add LRU caching

### PR DESCRIPTION
**Pull request by rphmeier**

### CODE REVIEW

**Comment by rphmeier** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/456?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#457** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/457?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#456** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/456?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#455** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/455?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rphmeier and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by gabriele-0201 — APPROVED** [review]

File: benchtop/src/workload.rs [hunk 1]
```diff
@@ -23,66 +24,155 @@ pub trait Workload: Send {
...
+        };
+        self.inner.run_step(&mut tx);
+    }
+
+    fn is_done(&self) -> bool {
+        self.inner.is_done()
+    }
+}
+
+struct LruCacheTransaction<'a> {
```
Comment (deuszx) [inline benchtop/src/workload.rs:151]: Aren't the lifetimes for `Transaction` and `LruCache` different? i.e. `LruCache` outlives the `Transaction. Do you use the same one for simplicity?

Comment (gabriele-0201) [reply benchtop/src/workload.rs:151]: I didn't write the code, so I could be wrong about the intention, but the main thing that the lifetime should provide to the compiler is that all references present in `LruCacheTransaction` must outlive the struct itself, and given its usage, they do. It doesn't really matter if `Transaction` of `LruCache` lives longer compared to the other, so I would say that for simplicity, they take the same lifetime

**Review by pepyakin — APPROVED** [review]

**Comment by pepyakin** [comment]
### Merge activity

* **Oct 14, 11:22 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/nomt/456).

File: benchtop/src/workload.rs (hunk 1 above)
Comment (rphmeier) [reply benchtop/src/workload.rs:151]: Yes, it's for simplicity!

What's important isn't the actual lifetimes of the objects, but their relationships to each other.

**Pull request merged**